python_scripts/one_time_scripts/refid.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+
+# This script updates caas_aspace_refid's based on a provided CSV.  For help on available arguments and options:
+# `python repeatable/update_fileuri.py -h`.
+import argparse
+import os
+import sys
+
+from dotenv import load_dotenv, find_dotenv
+from loguru import logger
+from pathlib import Path
+from python_scripts.utilities import client_login, read_csv
+
+# Logging
+logger.remove()
+log_path = Path(f'./logs', 'update_refid_{time:YYYY-MM-DD}.log')
+logger.add(str(log_path), format="{time}-{level}: {message}")
+
+# Find  and load environment-specific .env file
+env_file = find_dotenv(f'.env.{os.getenv("ENV", "dev")}')
+load_dotenv(env_file)
+
+def parseArguments():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("csvPath", help="path to csv input file", type=str)
+    parser.add_argument("-dR", "--dry-run", help="dry run?", action='store_true')
+    parser.add_argument("--version", action="version", version='%(prog)s - Version 1.0')
+
+    return parser.parse_args()
+    
+def get_refid(client, resource_uri):
+    existing_refid = client.get(f'plugins/caas_next_refid/find_by_uri?resource_uri={resource_uri}').json()
+    if 'error' in existing_refid:
+        logger.error(f'Refid will be created for {resource_uri}')
+        (f'Refid will be created for {resource_uri}')
+        return True
+    else:
+        logger.error(f'Refid already exists for {resource_uri}')
+        (f'Refid already exists for {resource_uri}')
+    
+def build_refid(obj):
+    refid = {}
+    refid['resource_uri'] = obj['uri']
+    refid['next_refid'] = obj['next_refid']
+
+    return refid
+
+def update_refid(client, data):
+    resource_uri = data['resource_uri']
+    next_refid = data['next_refid']
+    update_message = client.post(f'plugins/caas_next_refid/set_by_uri?resource_uri={resource_uri}&next_refid={next_refid}').json()
+    if 'error' in update_message:
+        logger.error(update_message)
+        print(f'ERROR: {update_message}')
+    else:
+        logger.info(f'{update_message}')
+        print(f'Updated object data: {update_message}')
```

Comment:
I haven't used it before, but I looked it up and we could use logger.success for successful posts to ASpace. I'll start using it in my scripts and see if it's useful.